services/validators.go

```diff
@@ -31,6 +31,18 @@ func ValidateJob(j models.JobSpec, store *store.Store) error {
 	return merr
 }
 
+// ValidateAdaptor checks that the bridge type doesn't have a duplicate name
+func ValidateAdaptor(bt *models.BridgeType, store *store.Store) error {
```

Comment:
Instead of having to read from the database every single time I would create an adapter name cache

Reply:
With bolt being a light weight key/value store, I don't think it'd provide much benefit. Especially with the rate that this endpoint may be called.